src/botPage/view/blockly/index.js

```diff
@@ -317,8 +317,40 @@ export default class _Blockly {
     load(blockStr = '', dropEvent = {}) {
         let xml;
 
+        try {
+            const xmlDoc = new DOMParser().parseFromString(blockStr, 'application/xml');
+
+            if (xmlDoc.getElementsByTagName('parsererror').length > 0) {
+                globalObserver.emit('ui.log.warn', `${translate('Unrecognized file format')}`);
+                return;
+            }
+        } catch (err) {
+            globalObserver.emit('ui.log.warn', `${translate('Unrecognized file format')}`);
+            return;
+        }
+
         try {
             xml = Blockly.Xml.textToDom(blockStr);
+            const blocklyXml = xml.querySelectorAll('block');
```

Comment:
Perhaps we could rename `blocklyXml` to `blockNodes` or equivalent so we know what's in the variable.